### PR TITLE
fix(ecs): removed explicit addition of ecs deployment type when circuit breaker is enabled

### LINKED ISSUE #16126: (aws-ecs): Adding circuitBreaker config causes service replacement if deploymentController was not previously set
When adding the circuitBreaker configuration to an ECS service, it automatically sets the `DeploymentController` to `ECS` in the resulting Cloudformation stack. If the `deploymentController` was not specified within the CDK previously, the default value of `ECS` is used when the service is created anyway. 

Unfortunately Cloudformation sees this as a change requiring replacement of the service.

### Reproduction Steps

Pre-existing service:
```
new FargateService(this, "Service", {
  cluster: cluster,
  taskDefinition: taskDefinition
})
```

Adding the circuitBreaker config:
```
new FargateService(this, "Service", {
  cluster: cluster,
  taskDefinition: taskDefinition,
  circuitBreaker: { enabled: true, rollback: true }
})
```
Deploying the above causes service replacement.

### What did you expect to happen?

Expected the `DeploymentController` to be left undefined, since that still results in the default value of `ECS` being used. 

### What actually happened?

Upon deployment the service required replacement, which will have been disruptive due to autoscaled values not carrying across to the new service.

Ended up using escape hatches to set the circuitBreak properties without adding the `DeploymentController` configuration.

### Environment

  - **CDK CLI Version  :** 1.118.0
  - **Framework Version:** 1.118.0
  - **Node.js Version:** v14.15.1 
  - **OS               :** Mac OS 11.5
  - **Language (Version):** Typescript (4.3.3) 

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by fergusdixon**
Fixes #16126

Copied from #16919 with review comments as it went stale and closed.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by fergusdixon** [comment]
Tests seem to be failing on unrelated deprecations:
I have tried to update the deprecated methods but they seem non-trivial?

```
@aws-cdk/aws-ecs: Summary of all failing tests
@aws-cdk/aws-ecs: FAIL test/external/external-service.test.js (8.896 s)
@aws-cdk/aws-ecs:   â— external service â€º add warning to annotations if circuitBreaker is specified with a non-ECS DeploymentControllerType
@aws-cdk/aws-ecs:     DeprecationError: @aws-cdk/aws-ecs.EcsOptimizedAmi is deprecated.
@aws-cdk/aws-ecs:       see {@link EcsOptimizedImage#amazonLinux}, {@link EcsOptimizedImage#amazonLinux} and {@link EcsOptimizedImage#windows}
@aws-cdk/aws-ecs:       This API will be removed in the next major release.
@aws-cdk/aws-ecs:       331 |       (options.machineImageType === MachineImageType.BOTTLEROCKET ? new BottleRocketImage({
@aws-cdk/aws-ecs:       332 |         architecture: options.instanceType.architecture,
@aws-cdk/aws-ecs:     > 333 |       }) : new EcsOptimizedAmi());
@aws-cdk/aws-ecs:           |            ^
@aws-cdk/aws-ecs:       334 |
@aws-cdk/aws-ecs:       335 |     const machineImageType = options.machineImageType ??
@aws-cdk/aws-ecs:       336 |       (isBottleRocketImage(machineImage) ? MachineImageType.BOTTLEROCKET : MachineImageType.AMAZON_LINUX_2);
@aws-cdk/aws-ecs:       at Cluster.addCapacity (lib/cluster.ts:333:12)
@aws-cdk/aws-ecs:       at Object.<anonymous> (test/external/external-service.test.ts:531:13)
@aws-cdk/aws-ecs: FAIL test/ec2/ec2-service.test.js (10.525 s)
@aws-cdk/aws-ecs:   â— ec2 service â€º When creating an EC2 Service â€º add warning to annotations if circuitBreaker is specified with a non-ECS DeploymentControllerType
@aws-cdk/aws-ecs:     DeprecationError: @aws-cdk/aws-ecs.EcsOptimizedAmi is deprecated.
@aws-cdk/aws-ecs:       see {@link EcsOptimizedImage#amazonLinux}, {@link EcsOptimizedImage#amazonLinux} and {@link EcsOptimizedImage#windows}
@aws-cdk/aws-ecs:       This API will be removed in the next major release.
@aws-cdk/aws-ecs:       331 |       (options.machineImageType === MachineImageType.BOTTLEROCKET ? new BottleRocketImage({
@aws-cdk/aws-ecs:       332 |         architecture: options.instanceType.architecture,
@aws-cdk/aws-ecs:     > 333 |       }) : new EcsOptimizedAmi());
@aws-cdk/aws-ecs:           |            ^
@aws-cdk/aws-ecs:       334 |
@aws-cdk/aws-ecs:       335 |     const machineImageType = options.machineImageType ??
@aws-cdk/aws-ecs:       336 |       (isBottleRocketImage(machineImage) ? MachineImageType.BOTTLEROCKET : MachineImageType.AMAZON_LINUX_2);
@aws-cdk/aws-ecs:       at Cluster.addCapacity (lib/cluster.ts:333:12)
@aws-cdk/aws-ecs:       at Object.<anonymous> (test/ec2/ec2-service.test.ts:1210:15)
@aws-cdk/aws-ecs: Test Suites: 2 failed, 20 passed, 22 total
```

File: packages/@aws-cdk/aws-ecs/lib/base/base-service.ts [hunk 1]
```diff
@@ -465,7 +463,9 @@ export abstract class BaseService extends Resource
     if (props.deploymentController?.type === DeploymentControllerType.EXTERNAL) {
       Annotations.of(this).addWarning('taskDefinition and launchType are blanked out when using external deployment controller.');
     }
-
+    if (props.circuitBreaker && props.deploymentController?.type !== DeploymentControllerType.ECS) {
+      Annotations.of(this).addWarning('Deployment circuit breaker requires the ECS deployment controller.');
```
Comment (corymhall) [inline packages/@aws-cdk/aws-ecs/lib/base/base-service.ts:467]: This should be `addError` if it is not allowed.

**Review by corymhall — DISMISSED** [review]
Just one minor comment, otherwise it looks good!

File: packages/@aws-cdk/aws-ecs/lib/base/base-service.ts (hunk 1 above)
Comment (fergusdixon) [reply packages/@aws-cdk/aws-ecs/lib/base/base-service.ts]: Done

**Review by corymhall — APPROVED** [review]

**Comment by rgoltz** [comment]
Hi @fergusdixon + @corymhall - Could you please check my comment here: https://github.com/aws/aws-cdk/issues/16126#issuecomment-1271626957 - Just to avoid issues or add needed clarification for existing stacks (while having a ServiceName defined).

**Pull request merged**